### DISCUSSION TITLE
how to install language models - medium model in my case on Mac M1 - without getting "SHA256 checksum does not match" ?

### DISCUSSION THREAD

**Discussion by skywalk-explore** [openai/whisper#2155] [Q&A] [answered]
The file upload up till 459 mb every time then I get the error. 

I've seen other post about this error, but no solutions were identified. One person said they just restarted something and the issue resolved it self. 

Any help ?

small.pt exists, but the SHA256 checksum does not match; re-downloading the file
  warnings.warn(
 99%|██████████████████████████████████████▊| 459M/461M [04:11<00:01, 1.91MiB/s]
Traceback (most recent call last):
  File "/Users/parenti/PycharmProject/srtTranscribe/main.py", line 40, in <module>
    transcribe_and_srt("Video/Frieren - 5.mov")
  File "/Users/parenti/PycharmProject/srtTranscribe/main.py", line 6, in transcribe_and_srt
    model = whisper.load_model("small")
  File "/Users/parenti/PycharmProject/srtTranscribe/venv/lib/python3.10/site-packages/whisper/__init__.py", line 133, in load_model
    checkpoint_file = _download(_MODELS[name], download_root, in_memory)
  File "/Users/parenti/PycharmProject/srtTranscribe/venv/lib/python3.10/site-packages/whisper/__init__.py", line 87, in _download
    raise RuntimeError(
RuntimeError: Model has been downloaded but the SHA256 checksum does not not match. Please retry loading the model.

**Comment by glangford** [comment]
Try 
`cd ~/.cache/whisper`

delete `small.pt` and try again

**Reply by glangford** [reply]
...if this still doesn't work, try running your test outside of pycharm or any other IDE. See also

* https://github.com/openai/whisper/discussions/1013

**Reply by skywalk-explore** [reply]
Prior to deleting model the first line of the error is the warning, as below.

- `1. /Users/xx/xxx/xx: UserWarning:    /Users/parenti/.cache/whisper/small.pt exists, but the SHA256 checksum does not match; re-downloading the file warnings.warn( 97%|██████████████████████████████████████▋ | 446M/461M [04:17<00:15, 982kiB/s])`


After deleting the small model, the warning begins with:

-  50%|███████████████████▍                   | 230M/461M [01:23<00:54, 4.45MiB/s]

then the rest of the error  is identical to the original.

It doesn't seem to work. Even after deleting I get the same error but with the above difference.

Went to the link provided and there I could download and use the medium model, it works but too big for Mac. Take 70 something minutes to transcribe a couple of minutes to mp3.

**Reply by skywalk-explore** [reply]
Having looked at the checksums of my file and the required file. there is a difference in the checksum. for some reason I don't have the correct file to the one required by the package. 

my small.pt file checksum: c39b7d8503417cf17df598db01964c50fff498b271d235f985d8bf90aab0c4ef

Checksum required by package: BzY8DmU6=0{>%Rpa?JkvJ6qF(V^F86#Xh7JUGMK}P<N0000

**Reply by skywalk-explore** [reply, marked as answer]
It's working now thanks man.

1. remove the small.pt from file using cd ~/.cache/whisper then rm small.pt

2. download the correct small.pt file from (https://github.com/openai/whisper/blob/ba3f3cd54b0e5b8ce1ab3de13e32122d0d5f98ab/whisper/__init__.py#L17)

3. copy the download small.pt file into the whisper cache. cp /Users/xxx/Downloads/small.pt ~/.cache/whisper/

Thanks again bro.
